dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNServiceManagerInterfaceIIDM.cpp

```diff
@@ -93,4 +98,104 @@ ServiceManagerInterfaceIIDM::getBusesConnectedBySwitch(const std::string& busId,
   return ret;
 }
 
+boost::shared_ptr<BusInterface>
+ServiceManagerInterfaceIIDM::getRegulatedBus(const std::string& regulatingComponent) const {
+  const auto& regulatingComp = dataInterface_->findComponent(regulatingComponent);
+  switch (regulatingComp->getType()) {
+    case ComponentInterface::GENERATOR: {
+      auto& generator = dataInterface_->getNetworkIIDM().getGenerator(regulatingComp.get()->getID());
+      return getRegulatedBusOnSide(generator.getRegulatingTerminal());
+    }
+    case ComponentInterface::SVC: {
+      auto& svc = dataInterface_->getNetworkIIDM().getStaticVarCompensator(regulatingComp.get()->getID());
+      return getRegulatedBusOnSide(svc.getRegulatingTerminal());
+    }
+    case ComponentInterface::SHUNT: {
+      auto& shunt = dataInterface_->getNetworkIIDM().getShuntCompensator(regulatingComp.get()->getID());
+      return getRegulatedBusOnSide(shunt.getRegulatingTerminal());
+    }
+    case ComponentInterface::UNKNOWN:
+    case ComponentInterface::BUS:
+    case ComponentInterface::TWO_WTFO:
+    case ComponentInterface::CALCULATED_BUS:
+    case ComponentInterface::SWITCH:
+    case ComponentInterface::LOAD:
+    case ComponentInterface::LINE:
+    case ComponentInterface::DANGLING_LINE:
+    case ComponentInterface::THREE_WTFO:
+    case ComponentInterface::VSC_CONVERTER:
+    case ComponentInterface::LCC_CONVERTER:
+    case ComponentInterface::HVDC_LINE:
```

Comment:
maybe add a comment to explain why only these 3 cases generate a regulated interface